### PR TITLE
NOD | Ensure API-loaded issues are in place

### PR DESCRIPTION
**Pull request by Mottie**
## Summary

- _(Summarize the changes that have been made to the platform)_
  > We noticed in staging that for freshly started forms, the save-in-progress prefill overwrites the contested issues set by the FormApp; this happens because the `useEffect` in FormApp is monitoring `formData.contestedIssues`, but the entire `formData` is replaced.
- _(If bug, how to reproduce)_
  > Start a new form in staging and navigate to the contestable issues page - no issues will load, and no alert will show stating that the API failed. Use the "finish this application later" link and then immediately continue the form, the contestable issues page will now show all the API-loaded issues
- _(What is the solution, why is this the solution)_
  > Contestable issues widget now checks if the form data value is undefined. 
- _(Which team do you work for, does your team own the maintenance of this component?)_
  > Benefits decision reviews 
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_

## Related issue(s)

[#67796](https://github.com/department-of-veterans-affairs/va.gov-team/issues/67796)

## Testing done

- _Describe what the old behavior was prior to the change_
  > Blank contestable issues page
- _Describe the steps required to verify your changes are working as expected_
  > See summary
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

N/A - no visual change

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/appeals/10182/components/ContestableIssuesWidget.jsx [hunk 1]
```diff
@@ -96,6 +100,24 @@ const ContestableIssuesWidget = props => {
...
+        ...formData,
+        contestedIssues: getEligibleContestableIssues(
+          contestableIssues?.issues,
+          {
+            showPart3,
+          },
+        ),
+      });
+    }
+  });
```
Comment (anniebtran) [inline src/applications/appeals/10182/components/ContestableIssuesWidget.jsx:119]: do we need to add the dependency array param here?

File: src/applications/appeals/10182/containers/FormApp.jsx [hunk 2]
```diff
@@ -3,7 +3,7 @@ import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 
 import RoutedSavableApp from 'platform/forms/save-in-progress/RoutedSavableApp';
-import { selectProfile, isLoggedIn } from 'platform/user/selectors';
```
[the remark this answers is not in the document]
Comment (Mottie) [reply src/applications/appeals/10182/containers/FormApp.jsx:6]: Yes, we're not using profile in the FormApp anymore.

File: src/applications/appeals/10182/components/ContestableIssuesWidget.jsx (hunk 1 above)
Comment (Mottie) [reply src/applications/appeals/10182/components/ContestableIssuesWidget.jsx:119]: I didn't add one, since it only needs to be called initially.

Comment (Mottie) [reply src/applications/appeals/10182/components/ContestableIssuesWidget.jsx:119]: Oops, I should have added an empty array :embarrassed:

**Review by anniebtran — APPROVED** [review]
LGTM

**Pull request merged**